src/transformStateWithClones.js

```diff
@@ -8,6 +8,51 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const statesHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = addProperties(currentState, action.extraData);
+        statesHistory.push(currentState);
+        break;
+      case 'removeProperties':
+        currentState = removeProperties(currentState, action.keysToRemove);
+        statesHistory.push(currentState);
+        break;
+      case 'clear':
+        currentState = clearProperties(currentState);
+        statesHistory.push(currentState);
```

Comment:
[DONT REPEAT YOURSELF]: You are pushing `currentState` to `statesHistory` in every case of the switch. This action should be done once after the switch block to avoid repetition.